pkg/controller/registry/resolver/querier_test.go

```diff
@@ -151,23 +165,52 @@ func TestNamespaceSourceQuerier_FindProvider(t *testing.T) {
 			},
 			args: args{
 				api: opregistry.APIKey{"group", "version", "kind", "plural"},
+				catalogKey: CatalogKey{},
 			},
 			out: out{
 				bundle: nil,
 				key:    nil,
 				err:    fmt.Errorf("group/version/kind (plural) not provided by a package in any CatalogSource"),
 			},
 		},
+		{
+			fields: fields{
+				sources: sources,
+			},
+			args: args{
+				api: opregistry.APIKey{"group2", "version2", "kind2", "plural2"},
+				catalogKey: CatalogKey{Name: "test2", Namespace: "ns"},
+			},
+			out: out{
+				bundle: bundle2,
+				key:    &CatalogKey{Name: "test2", Namespace: "ns"},
+				err:    nil,
+			},
+		},
+		{
+			fields: fields{
+				sources: sources,
+			},
+			args: args{
+				api: opregistry.APIKey{"group2", "version2", "kind2", "plural2"},
+				catalogKey: CatalogKey{Name: "test3", Namespace: "ns"},
+			},
+			out: out{
+				bundle: bundle2,
+				key:    &CatalogKey{Name: "test2", Namespace: "ns"},
+				err:    nil,
+			},
+		},
```

Comment:
I think we should have a test to ensure that the correct bundle is returned when more than one source containing a provider is given along with the preferred source key.

Reply:
This is covered in the E2E testcase. I can create a Unit test as well.

Reply:
It's probably best to add it to `master`.